File: packages/core/rules/test/excessive-agency/negative.py
# Test cases that should NOT be detected (false positives)

import os
import subprocess
import shutil

# File deletion with approval
if confirm("Delete this file?"):
    os.remove(file_path)

# File deletion with user approval
user_approval("delete-file")
shutil.rmtree(dir_path)

# Code execution with approval
approved = ask_user("Execute command?")
if approved:
    os.system(user_command)

# File write with path restriction
if user_path.startswith(ALLOWED_DIR):
    with open(user_path, "w") as f:
        f.write(user_data)

# File write with validation
if is_allowed_path(user_path):
    with open(user_path, "w") as f:
        f.write(user_data)

File: packages/core/rules/test/excessive-agency/negative.js
// Test cases that should NOT be detected (false positives)

const fs = require('fs');
const { exec, spawn } = require('child_process');

// File deletion with approval
if (await confirm("Delete this file?")) {
  fs.unlinkSync(filePath);
}

// File deletion with user approval
await userApproval("delete-file");
fs.rm(dirPath, { recursive: true });

// Code execution with approval
const approved = await askUser("Execute command?");
if (approved) {
  exec(userCommand);
}

// Network access with URL validation
async function myTool(url) {
  if (isAllowedUrl(url)) {
    const data = await fetch(url);
    return data;
  }
}

// File write with path restriction
if (userPath.startsWith(ALLOWED_DIR)) {
  fs.writeFileSync(userPath, userData);
}

// File write with validation
if (isAllowedPath(userPath)) {
  fs.writeFileSync(userPath, userData);
}
